### PR TITLE
[Hotfix] Dedupe NotificationSubscriptions [OSF-7782]

### PR DESCRIPTION
**Pull request by mfraezz**
## Purpose
Dedupe `NotificationSubscription` by `_id` and prevent the creation of additional dupes

## Changes
* Migration to remove dupes, add uniqueness index
* Use `get_or_create` instead of just creating

## Side effects
Migrations in `develop` will need a bump/merge

## Ticket
[[OSF-7782]](https://openscience.atlassian.net/browse/OSF-7782)

### CODE REVIEW

File: osf/migrations/0021_unique_notificationsettings__ids.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+# This is an auto-migration and not a management command because:
+#   1. The next script would fail if duplicate records existed
+#   2. This should only need to be ran once
```
Comment (cwisecarver) [inline osf/migrations/0021_unique_notificationsettings__ids.py]: run* once

File: website/notifications/utils.py [hunk 2]
```diff
@@ -428,8 +428,9 @@ def subscribe_user_to_global_notifications(user):
     for user_event in user_events:
         user_event_id = to_subscription_key(user._id, user_event)
 
-        subscription = NotificationSubscription(_id=user_event_id, owner=user, event_name=user_event)
-        subscription.save()  # Need to save in order to access m2m fields
+        subscription, created = NotificationSubscription.objects.get_or_create(_id=user_event_id, user=user, event_name=user_event)
+        if created:
+            subscription.save()  # Need to save in order to access m2m fields
```
Comment (cwisecarver) [inline website/notifications/utils.py]: I'm 99% sure `get_or_create` saves the object.
[Yup](https://docs.djangoproject.com/en/1.11/ref/models/querysets/#get-or-create)

File: website/notifications/utils.py [hunk 3]
```diff
@@ -428,8 +428,9 @@ def subscribe_user_to_global_notifications(user):
...
         subscription.add_user_to_subscription(user, notification_type)
         subscription.save()
```
Comment (cwisecarver) [inline website/notifications/utils.py]: You didn't do this, but `add_user_to_subscription` has a `save=True` default kwarg so this save is unnecessary.

Comment (mfraezz) [inline website/notifications/utils.py]: ¯\\\_(ツ)_/¯

**Comment by cwisecarver** [comment]
:shipit:

**Pull request merged**